docs/demultiplexing.md

```diff
@@ -0,0 +1,174 @@
+
+![diagram](demultiplexer.svg)
+
+Example configuration file with one demultiplexer:
+```
+{
+    "model_config_list": [
+        {"config": {
+            "name": "Model_A",
+            "base_path": "/models/model_a"
+        }},
+        {"config": {
+            "name": "Model_B",
+            "base_path": "/models/model_b"
+        }},
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "my_pipeline_with_demultiplexer",
+            "inputs": ["pipeline_input_name"],
+            "nodes": [
+                {
+                    "name": "Model_A_node",
+                    "model_name": "Model_A",
+                    "type": "DL model",
+                    "demultiply_count": 3,  <------------------ this parameter specifies how many branches should be spawned
+                    "inputs": [
+                        {"input": {"node_name": "request",
+                                   "data_item": "pipeline_input_name"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output_A",
+                         "alias": "output_A"},
+                        {"data_item": "output_B",
+                         "alias": "output_B"}
+                    ]
+                },
+                {
+                    "name": "Model_B_node",
+                    "model_name": "Model_B",
+                    "type": "DL model",
+                    "inputs": [
+                        {"input_A": {"node_name": "Model_A_node",
+                                     "data_item": "output_A"}},
+                        {"input_B": {"node_name": "Model_A_node",
+                                     "data_item": "output_B"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "output",
+                         "alias": "output"}
+                    ]
+                }
+            ],
+            "outputs": [
+                {"pipeline_output_name": {"node_name": "Model_B_node",
+                                          "data_item": "output"}}
+            ]
+        }
+    ]
+}
+```
+
+## Dynamic demultiply_count parameter
+There might be use cases where one custom node library is used to produce unpredictable number of batch. To achieve it, `demultiply_count` can be set to `0`. This indicates that pipeline supports any number of batch returned by custom node: `(X,N,C,H,W,...)` - where `X` is dynamic `demultiply_count`. OpenVINO&trade; Model Server is capable of interpreting such dynamic batch and is able to split outputs into dynamic number of pipeline branches. When using dynamic `demultiply_count` parameters, only one demultiplexer can exist in pipeline. Important to note - when batch 0 is returned, pipeline stops its execution and reports an error to gRPC/REST response.
+
```

Comment:
stops execution and returns no content response.

Reply:
Done